### PR TITLE
editoast: fix infra update

### LINKED ISSUE #4108: Editing an infra modifies additional fields (locked, version..)
### What happened?

Non modified fields are modified when an infra is edited.
The issue is due to the infra model : https://github.com/DGEXSolutions/osrd/blob/dev/editoast/src/models/infra.rs
Some fields which should be optional, are not. Whenever any of the infra's fields are updated, it also changes these none optional fields to their default value (for example, locked is changed to its default value, false).

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

Open the Infra list.
Edit the infra (for example its name).
The infra becomes unlocked, its version changes to 0 etc.

### What operating system, browser and environment are you using?

All

### OSRD version (top right corner `Account` button > `Informations`)

e9332e9aa

### PR DESCRIPTION
**Pull request by Erashin**
Remove default values in infra model (except modified_date which must be always modified) and replace them with Option. Set default values at creation. Default values will not be set at update anymore.

Fixes #4108.

### CODE REVIEW

File: editoast/openapi.yaml [hunk 1]
```diff
@@ -187,6 +187,13 @@ paths:
       tags:
         - infra
       summary: Create an infra
+      parameters:
+        - in: query
+          name: locked
+          description: whether or not the infra can be edited/deleted.
+          default: false
+          schema:
+            type: bool
```
Comment (flomonster) [inline editoast/openapi.yaml]: We don't need to specify if an empty infra should be locked or not.

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks for this bug fix. Except for the unnecessary query param, everything looks good to me.

File: editoast/openapi.yaml (hunk 1 above)
Comment (Erashin) [reply editoast/openapi.yaml]: 100%. Confusion with my other PR. Will get rid of this.

**Review by clarani — COMMENTED** [review]
LGTM

**Review by leovalais — APPROVED** [review]

**Review by flomonster — APPROVED** [review]
LGTM. Thanks for this bug fix :tada:

**Pull request merged**